### PR TITLE
Partner author image validation

### PR DESCRIPTION
**Pull request by tomneeman151293**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/39173

## Description
Adding validation to author image. The rules defined by @ShahafBenYakir were
1) if image exists - validate image is not empty
2) if support level of the pack is partner - make sure such image exists

Note: currently I merged my other branch into this one, https://github.com/demisto/demisto-sdk/pull/1480, because I added an enhancement in packs unique file to read metadata only once, and since my new addition was using pack metadata in same file, I merged https://github.com/demisto/demisto-sdk/pull/1480 branch into mine. After https://github.com/demisto/demisto-sdk/pull/1480 will be merged to master the changes will appear to be much smaller.
## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Comment by tomneeman151293** [comment]
@ShahafBenYakir FYI, 
**partners without author image:**
- Packs/AlphaSOC_Network_Behavior_Analytics
- Packs/AlphaSOC_Wisdom
- Packs/DomainTools
- Packs/Ironscales
- Packs/PhishAI
- Packs/SCADAfence_CNM
- Packs/Skyformation

**Pack with a larger than 1MB image:**
- Packs/Cortex911

**Comment by ShahafBenYakir** [comment]
@kgal-pan Please have a look on the above 👆

**Comment by kgal-pan** [comment]
@tomneeman151293 @ShahafBenYakir - 

See comments below.

Regarding:

> Packs/AlphaSOC_Network_Behavior_Analytics
> Packs/AlphaSOC_Wisdom
> Packs/DomainTools
> Packs/PhishAI
> Packs/SCADAfence_CNM
> Packs/Skyformation

These packs has a non-standard folder structure, have missing Python source code and are missing an image also in the integration folder, e.g.:

```bash
tree Packs/AlphaSOC_Network_Behavior_Analytics
Packs/AlphaSOC_Network_Behavior_Analytics
├── Integrations
│   ├── integration-AlphaSOC_Network_Behavior_Analytics.yml
│   └── integration-AlphaSOC_Network_Behavior_Analytics_README.md
├── README.md
├── ReleaseNotes
│   ├── 1_0_1.md
│   └── 1_0_2.md
└── pack_metadata.json
```

What should we do with them?

---

Regarding:

> Packs/Ironscales

Run command below to fix:
```bash
cp Packs/Ironscales/Integrations/Ironscales/Ironscales_image.png Packs/Ironscales/Author_image.png
```

---

Regarding:

> Pack with a larger than 1MB image:
> Packs/Cortex911

This Pack doesn't have any Cortex XSOAR resources or source code. From the `README`, looks like it was created by the sales team but not sure what's the purpose of this pack and if it needs to be fixed.

**Comment by ShahafBenYakir** [comment]
@tomneeman151293 Please ignore this validation in Cortex911 the pack-ignore file.

**Comment by kgal-pan** [comment]
@tomneeman151293 @ShahafBenYakir 
7 missing Partner author images added, [link to PR](https://github.com/demisto/content/pull/13836)

File: demisto_sdk/commands/common/hook_validations/image.py [hunk 1]
```diff
@@ -61,10 +61,20 @@ def is_valid(self):
...
+    def has_valid_size(self, allow_empty_image_file: bool, maximum_size: int = IMAGE_MAX_SIZE) -> None:
+        """
+        Checks if image has a valid size.
+        if 'allow_empty_image_file' is true, checks that the image file is not empty.
+        Args:
+            allow_empty_image_file (bool): Whether empty image file is an error.
+            maximum_size (int): Maximum allowed size.
+
+        Returns:
+            (bool): True if size is valid, false otherwise.
```
Comment (dantavori) [inline demisto_sdk/commands/common/hook_validations/image.py:73]:
```suggestion
```

File: demisto_sdk/commands/common/hook_validations/image.py [hunk 2]
```diff
@@ -61,10 +61,20 @@ def is_valid(self):
 
         return self._is_valid
 
-    def oversize_image(self):
-        """Check if the image if over sized, bigger than IMAGE_MAX_SIZE"""
+    def has_valid_size(self, allow_empty_image_file: bool, maximum_size: int = IMAGE_MAX_SIZE) -> None:
```
Comment (dantavori) [inline demisto_sdk/commands/common/hook_validations/image.py:64]:
```suggestion
    def validate_size(self, allow_empty_image_file: bool, maximum_size: int = IMAGE_MAX_SIZE) -> None:
```
best practice: don't call a function `is_<something>`, `has_<something>`, etc., if it doesn't return a boolean value. either return self._is_valid and use it when you call the function, or change the name as suggested above.

File: demisto_sdk/commands/common/tests/pack_metadata_validator_test.py [hunk 3]
```diff
@@ -96,3 +96,19 @@ def test_name_does_not_contain_excluded_word(self):
     def read_file(file_):
         with io.open(file_, mode="r", encoding="utf-8") as data:
             return data.read()
+
+    def test_metadata_not_dict(self, mocker):
```
Comment (dantavori) [inline demisto_sdk/commands/common/tests/pack_metadata_validator_test.py:100]: why is this relevant to the PR?

File: CHANGELOG.md [hunk 4]
```diff
@@ -6,6 +6,7 @@
 * Expanded the validations performed on beta integrations.
 * Improved the error messages in **generate-docs**, if an example was not provided.
 * Added to **validate** command a validation that a content entity or a pack name does not contain the words "partner" and "community".
+* Added to the **validate** command validation for author image.
```
Comment (dantavori) [inline CHANGELOG.md:9]:
```suggestion
* Added to the **validate** command a validation for the author image.
```

File: demisto_sdk/commands/common/errors.py [hunk 5]
```diff
@@ -851,6 +853,18 @@ def invalid_image_name():
         return "The image's file name is invalid - " \
                "make sure the name looks like the following: <integration_name>_image.png"
 
+    @staticmethod
+    @error_code_decorator
+    def image_is_empty(image_path: str):
+        return f'Image corresponding to image path {image_path} should not be empty for partner support level. ' \
```
Comment (dantavori) [inline demisto_sdk/commands/common/errors.py:859]:
```suggestion
        return f'The author image in path {image_path} should not be empty. ' \
```
correct me if I'm wrong, not only for partners

File: demisto_sdk/commands/common/errors.py [hunk 6]
```diff
@@ -851,6 +853,18 @@ def invalid_image_name():
...
+               'Please provide a relevant image.'
+
+    @staticmethod
+    @error_code_decorator
+    def author_image_is_missing(image_path: str):
+        return f'Image corresponding to image path {image_path} should exist for partner support level. Please ' \
+               'provide a relevant image'
```
Comment (dantavori) [inline demisto_sdk/commands/common/errors.py:866]:
```suggestion
        return f'Partners must provide a non-empty author image under the path {image_path}. '
```

**Review by dantavori — CHANGES_REQUESTED** [review]

File: demisto_sdk/commands/common/hook_validations/image.py (hunk 2 above)
Comment (tomneeman151293) [reply demisto_sdk/commands/common/hook_validations/image.py:64]: Noted

File: demisto_sdk/commands/common/tests/pack_metadata_validator_test.py (hunk 3 above)
Comment (tomneeman151293) [reply demisto_sdk/commands/common/tests/pack_metadata_validator_test.py:100]: Added for coverage, saw a line that wasn't covered and added it

File: demisto_sdk/commands/common/errors.py (hunk 5 above)
Comment (tomneeman151293) [reply demisto_sdk/commands/common/errors.py:859]: Correct. Fixing

**Review by dantavori — APPROVED** [review]
